opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchQueryRequest.java

```diff
@@ -203,12 +205,23 @@ public OpenSearchResponse searchWithPIT(Function<SearchRequest, SearchResponse>
       if (searchAfter != null) {
         this.sourceBuilder.searchAfter(searchAfter);
       }
-      // Set sort field for search_after
-      if (this.sourceBuilder.sorts() == null) {
+      // Add sort tiebreaker for PIT search.
+      // We cannot remove it since `_shard_doc` is not added implicitly in PIT now.
+      // Ref https://github.com/opensearch-project/OpenSearch/pull/18924#issuecomment-3342365950
+      if (this.sourceBuilder.sorts() == null || this.sourceBuilder.sorts().isEmpty()) {
+        // If no sort field specified, sort by `_doc` + `_shard_doc`to get better performance
         this.sourceBuilder.sort(DOC_FIELD_NAME, ASC);
-        // Workaround to preserve sort location more exactly,
-        // see https://github.com/opensearch-project/sql/pull/3061
-        this.sourceBuilder.sort(METADATA_FIELD_ID, ASC);
+        this.sourceBuilder.sort(SortBuilders.shardDocSort());
```

Comment:
Does it matter if we duplicate fields in the sorting list? We could simplify/remove the below `else` logic by just always appending this, I would expect Lucene to optimize it in the background but I haven't measured it.